tools/generate-parser/index.ts

```diff
@@ -128,7 +129,11 @@ async function main() {
         }
     }
 
-    const generator = new SLRParserGenerator(grammar, 'input', typeMap['input'] || 'any');
-    await writeout(firstFile.preamble, generator, fs.createWriteStream(output), output, typeMap['input'] || 'any');
+    if (output === '--wsn') {
+        wsnDump(grammar);
```

Comment:
what does `wsn` stand for?

Reply:
https://en.wikipedia.org/wiki/Wirth_syntax_notation

It's an alternative to BNF, and it's one of the input formats of [kgt](https://github.com/katef/kgt), the tool I found to generate the documentation for the grammar.